payments_service/solitude/__init__.py

```diff
@@ -0,0 +1,70 @@
+from django.conf import settings
+
+from curling.lib import API
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from slumber.exceptions import HttpClientError
+
+from ..base.views import error_400, error_405
+
+
+def api():
+    # TODO: add oauth.
+    return API(settings.SOLITUDE_URL)
+
+
+class SolitudeBodyguard(APIView):
+    """
+    API view that proxies a single call downstream to Solitude
+    with some restrictions.
+
+    In other words, it's like Solitude's bodyguard because
+    Solitude is firewalled off from direct Internet connections.
+    """
+    # A list of allowed methods, as lowercase Curling properties.
+    # Example: methods=['get', 'post']
+    methods = []
+    # Path to a resource, as a dot-separated Curling attribute path.
+    # Example:
+    # GET /services/status/
+    # would be:
+    # resource = 'services.status'
+    resource = None
+
+    def _api_request(self, method, *args, **kw):
+        if method.lower() not in self.methods:
+            return error_405()
+
+        # Get the endpoint + method, such as api.services.status.get
+        api_request = getattr(self._resource(), method)
+
+        try:
+            result = api_request(*args, **kw)
```

Comment:
I thought about statsd but solitude already has timing so I'm not sure how useful it would be. Maybe it could pinpoint a service connectivity issue rather than a solitude one I suppose.